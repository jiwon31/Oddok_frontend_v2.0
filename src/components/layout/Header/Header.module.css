.header {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1.5rem 2.5rem;
  background-color: var(--gray950);
}

.logo,
.pages,
.nav {
  flex: 1;
}

.logo {
  font-size: 1.5rem;
  font-weight: bold;
  text-decoration-line: none;
  color: var(--gray300);
}

.pages {
  text-align: center;
}

.page {
  margin: 0 1rem;
  font-size: 1.1rem;
  font-weight: 500;
  text-decoration-line: none;
  color: var(--gray600);
}

.page.clicked {
  color: white;
}

.page:hover {
  color: white;
}

.nav {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  gap: 1rem;
}

.button {
  padding: 0.7rem 1rem;
  font-size: 0.8rem;
  font-weight: 500;
  text-decoration-line: none;
  border-radius: 4px;
  color: var(--gray950);
  background-color: var(--primary);
}
