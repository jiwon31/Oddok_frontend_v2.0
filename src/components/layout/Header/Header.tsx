import { Link, useLocation } from "react-router-dom";
import { Search } from "assets/icons";
// import UserInfo from "../UserInfo/UserInfo";
import styles from "./Header.module.css";

function Header() {
  const { pathname } = useLocation();

  return (
    <header className={styles.header}>
      <Link to="/" className={styles.logo}>
        ODDOK
      </Link>
      <div className={styles.pages}>
        <Link to="/" className={`${styles.page} ${pathname === "/" && styles.clicked}`}>
          스터디룸
        </Link>
        <Link to="/mypage" className={`${styles.page} ${pathname === "/mypage" && styles.clicked}`}>
          마이페이지
        </Link>
      </div>
      <nav className={styles.nav}>
        <Link to="/search" className={styles.search}>
          <Search />
        </Link>
        {/* <UserInfo /> */}
        <Link to="/studyroom/create" className={styles.button}>
          + 새 스터디 만들기
        </Link>
      </nav>
    </header>
  );
}
export default Header;
