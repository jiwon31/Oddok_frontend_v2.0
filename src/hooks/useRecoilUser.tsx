import { useRecoilState } from "recoil";
import { userState } from "recoil/user-state";

export default function useRecoilUser() {
  const [user, setUser] = useRecoilState(userState);

  return { user, setUser };
}
