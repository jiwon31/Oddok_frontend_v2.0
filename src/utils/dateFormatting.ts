// Date 객체를 yyyy-mm-dd 형식의 문자열로 반환
export default function dateFormatting(date: Date): string {
  const year = date.getFullYear();
  const month = date.getMonth() + 1;
  const day = date.getDate();
  return `${year}-${month < 10 ? `0${month}` : month}-${day < 10 ? `0${day}` : day}`;
}
