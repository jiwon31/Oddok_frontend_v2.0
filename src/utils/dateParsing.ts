// yyyy-mm-dd 형태의 문자열을 Date 객체로 변환
export default function dateParsing(value: string): Date {
  const date = new Date(value); // 로컬 타임존이 반영된 시간
  const offset = date.getTimezoneOffset() * 60000;
  return new Date(date.getTime() + offset);
}
