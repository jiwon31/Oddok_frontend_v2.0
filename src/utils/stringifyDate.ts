export default function stringifyDate(date: Date): string {
  const offset = date.getTimezoneOffset() * 60000;
  return new Date(+date - offset).toISOString();
}
