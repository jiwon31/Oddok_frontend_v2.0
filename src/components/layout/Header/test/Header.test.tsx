import { render, screen } from "@testing-library/react";
import useRecoilUser from "hooks/useRecoilUser";
import { Route } from "react-router-dom";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import { fakeUser } from "tests/user";
import { withRecoil, withRouter } from "tests/utils";
import Header from "../Header";

jest.mock("hooks/useRecoilUser");
const useRecoilUserMock = useRecoilUser as jest.MockedFunction<typeof useRecoilUser>;

describe("Header", () => {
  it("renders when a user is not logged in", () => {
    useRecoilUserMock.mockImplementation(() => ({ user: null, setUser: jest.fn() }));
    snapshot();
  });

  it("renders when a user is logged in", () => {
    useRecoilUserMock.mockImplementation(() => ({ user: fakeUser, setUser: jest.fn() }));
    snapshot();
  });

  it("should go to mypage when a user logged in clicks 마이페이지 button", async () => {
    useRecoilUserMock.mockImplementation(() => ({ user: fakeUser, setUser: jest.fn() }));
    render(
      withRouter(
        <>
          <Route path="/" element={<Header />} />
          <Route path="/mypage" element={<p>mypage</p>} />
        </>,
      ),
    );

    userEvent.click(screen.getByText("마이페이지"));
    expect(await screen.findByText(/mypage/i)).toBeInTheDocument();
  });

  it("should go to studyroom create page when a user logged in clicks 새 스터디 만들기 button", async () => {
    useRecoilUserMock.mockImplementation(() => ({ user: fakeUser, setUser: jest.fn() }));
    render(
      withRouter(
        <>
          <Route path="/" element={<Header />} />
          <Route path="/studyroom/create" element={<p>create new studyroom</p>} />
        </>,
      ),
    );

    userEvent.click(screen.getByText(/새 스터디 만들기/i));
    expect(await screen.findByText(/create new studyroom/i)).toBeInTheDocument();
  });

  function snapshot() {
    const component = renderer.create(withRecoil(withRouter(<Route path="/" element={<Header />} />)));
    expect(component.toJSON()).toMatchSnapshot();
  }
});
