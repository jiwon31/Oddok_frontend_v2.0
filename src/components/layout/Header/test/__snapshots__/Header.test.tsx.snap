// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders when a user is logged in 1`] = `
<header
  className="header"
>
  <a
    className="logo"
    href="/"
    onClick={[Function]}
  >
    ODDOK
  </a>
  <div
    className="pages"
  >
    <a
      className="page clicked"
      href="/"
      onClick={[Function]}
    >
      스터디룸
    </a>
    <a
      className="page false"
      href="/mypage"
      onClick={[Function]}
    >
      마이페이지
    </a>
  </div>
  <nav
    className="nav"
  >
    <a
      className="search"
      href="/search"
      onClick={[Function]}
    >
      <svg>
        search.svg
      </svg>
    </a>
    <div
      className="user_info"
    >
      <button
        className="profile"
        onClick={[Function]}
        type="button"
      >
        <svg>
          person-circle.svg
        </svg>
        <span
          className="nickname"
        >
          user
        </span>
      </button>
    </div>
    <a
      className="button"
      href="/studyroom/create"
      onClick={[Function]}
    >
      + 새 스터디 만들기
    </a>
  </nav>
</header>
`;

exports[`Header renders when a user is not logged in 1`] = `
<header
  className="header"
>
  <a
    className="logo"
    href="/"
    onClick={[Function]}
  >
    ODDOK
  </a>
  <div
    className="pages"
  >
    <a
      className="page clicked"
      href="/"
      onClick={[Function]}
    >
      스터디룸
    </a>
    <a
      className="page false"
      href="/mypage"
      onClick={[Function]}
    >
      마이페이지
    </a>
  </div>
  <nav
    className="nav"
  >
    <a
      className="search"
      href="/search"
      onClick={[Function]}
    >
      <svg>
        search.svg
      </svg>
    </a>
    <a
      className="button"
      href="/studyroom/create"
      onClick={[Function]}
    >
      + 새 스터디 만들기
    </a>
  </nav>
</header>
`;
